runtime/include/chpl-mem-array.h

```diff
@@ -54,6 +55,15 @@ static inline
 void* chpl_mem_array_alloc(size_t nmemb, size_t eltSize,
                            c_sublocid_t subloc, chpl_bool* callPostAlloc,
                            int32_t lineno, int32_t filename) {
+  void* p = NULL;
+#ifdef HAS_GPU_LOCALE
+  if (chpl_gpu_running_on_gpu_locale()) {
+    *callPostAlloc = false;
+    p = chpl_gpu_mem_alloc(nmemb*eltSize, CHPL_RT_MD_ARRAY_ELEMENTS,
+                           lineno, filename);
+  }
```

Comment:
Note that as-implemented this skips our malloc hooks (which means we won't track memory allocations for GPU memory)

I'd also probably hoist the size computation from below above and use that here instead of doing the math again (not that it's hard math, but for uniformity in calls)

I think you'll also want to implement realloc for arrays too. I'm not sure that it's needed today since I think the module code is protected by a `LOCALE_MODEL=="flat"` check, but it'd be good to implement this code or put in assert so it's easier to catch if that support is extended in the future.

Reply:
I agree

Reply:
> Note that as-implemented this skips our malloc hooks (which means we won't track memory allocations for GPU memory)

Yep.  I want to settle on https://github.com/Cray/chapel-private/issues/2404 before we make that connection, if that makes sense.

The rest, I'll fix.